src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const newObject: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemKey = `${item[key]}`;
+
+    if (!(itemKey in newObject)) {
```

Comment:
more safely to use hasOwnProperty

[article](https://masteringjs.io/tutorials/fundamentals/hasownproperty)